src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAreasOfLawServiceTest.java

```diff
@@ -78,4 +98,21 @@ void shouldReturnNotFoundWhenRetrievingCourtLocalAuthoritiesForNonExistentCourt(
             .isInstanceOf(NotFoundException.class)
             .hasMessage(NOT_FOUND + COURT_SLUG);
     }
+
+    @Test
+    void shouldUpdateCourtAreasOfLaw() throws IOException {
+        final String expectedJson = getResourceAsJson(TEST_COURT_AREAS_OF_LAW_PATH);
+        final List<AreaOfLaw> areasOfLaw = asList(OBJECT_MAPPER.readValue(expectedJson, AreaOfLaw[].class));
+
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtAreaOfLawRepository.saveAll(any())).thenReturn(COURT_AREA_OF_LAWS);
+
+        List<AreaOfLaw> courtAreasOfLawResult = adminCourtAreasOfLawService.updateAreasOfLawForCourt(COURT_SLUG, areasOfLaw);
+        verify(courtAreaOfLawRepository).deleteCourtAreaOfLawByCourtId(court.getId());
+        verify(courtAreaOfLawRepository).saveAll(anyIterable());
+
+        assertThat(courtAreasOfLawResult)
+            .hasSize(COURT_AREAS_OF_LAW_COUNT)
+            .containsExactlyElementsOf(areasOfLaw);
+    }
 }
```

Comment:
Can add a test here for court slug not found

Reply:
Added one for updating court areas of law, when slug not found for court